ts/tree/tree.ts

```diff
@@ -29,3 +30,23 @@ export interface Tree {
     updateBatch(offset: number, leaves: Array<Node>): void;
     isZero(level: number, leafIndex: number): boolean;
 }
+
+export interface AsyncTree {
```

Comment:
change memory tree to use promises but use same interface for both memory and db tree

Reply:
keeping independent Async Tree to not mess with Memory Tree